MREUnityRuntime/MREUnityRuntimeLib/IPC/Connections/WebSocket.cs

```diff
@@ -101,6 +103,55 @@ private async Task CloseInternal(bool clean, string reason)
 			}
 		}
 
+		/// <summary>
+		/// Sends a message to ther server, doing JSON serialization off thread. Moving this work
+		/// off thread saved us 30ms on several large frames in the chess app on a high end CPU. Note that
+		/// Some of the messages use enumerables to pass collections, which will now be executed off
+		/// thread. This is a performance win in many cases, but it means we can no longer access unity
+		/// objects through these enumerators.
+		/// </summary>
+		public void Send(Message message)
+		{
+			if (disposed)
+			{
+				throw new ObjectDisposedException("WebSocket has been disposed");
+			}
+
+			if (_ws.State != WebSocketState.Open)
+			{
+				throw new InvalidOperationException("WebSocket is not open");
+			}
+
+			// Ensure the message is sent on this websocket rather than a websocket allocated in the future (in case of a reconnect).
+			var ws = _ws;
+
+			_sendQueue.Add(async () =>
+			{
+				try
+				{
+					message.Id = Guid.NewGuid().ToString();
+					var json = JsonConvert.SerializeObject(message, Constants.SerializerSettings);
+
+					if (MREAPI.AppsAPI.VerboseLogging)
+					{
+						MREAPI.Logger.LogDebug($"Send: {json}");
+					}
+
+					var buffer = new ArraySegment<byte>(Encoding.UTF8.GetBytes(json));
+					try
+					{
+						await ws.SendAsync(buffer, WebSocketMessageType.Text, true, CancellationToken).ConfigureAwait(false);
+					}
+					catch (Exception)
+					{ }
+				}
+				catch (Exception e)
+				{
+					MREAPI.Logger.LogDebug($"Error serializing message. Exception: {e.Message}\nStackTrace: {e.StackTrace}");
+				}
+			});
+		}
+
 		/// <inheritdoc />
 		public void Send(string message)
```

Comment:
>Send [](start = 14, length = 4)

WebSocket changes look fine. I'd like to reimplement them in a separate branch and make some collateral changes, like removing `Send(string)`. Also, I'd like to test this change in isolation.